fluentgen/src/main/java/com/azure/autorest/fluent/FluentGen.java

```diff
@@ -0,0 +1,169 @@
+
+            codeModel = transformer.postTransform(codeModel);
+
+            // Step 3: Map
+            logger.info("Map code model to client model");
+            new FluentMapper(fluentJavaSettings).preModelMap(codeModel);
+
+            Client client = Mappers.getClientMapper().map(codeModel);
+
+            // Step 4: Write to templates
+            logger.info("Java template for client model");
+            JavaPackage javaPackage = new JavaPackage();
+            // Service client
+            javaPackage.addServieClient(client.getServiceClient().getPackage(), client.getServiceClient().getClassName(), client.getServiceClient());
+            if (JavaSettings.getInstance().shouldGenerateClientInterfaces()) {
+                javaPackage.addServiceClientInterface(client.getServiceClient().getInterfaceName(), client.getServiceClient());
+            }
+
+            // Service client builder
+            javaPackage.addServieClientBuilder(client.getServiceClient().getPackage(), client.getServiceClient().getInterfaceName() + "Builder", client.getServiceClient());
+
+            // Method group
+            for (MethodGroupClient methodGroupClient : client.getServiceClient().getMethodGroupClients()) {
+                javaPackage.addMethodGroup(methodGroupClient.getPackage(), methodGroupClient.getClassName(), methodGroupClient);
+                if (JavaSettings.getInstance().shouldGenerateClientInterfaces()) {
+                    javaPackage.addMethodGroupInterface(methodGroupClient.getInterfaceName(), methodGroupClient);
+                }
+            }
+
+            // Response
+            for (ClientResponse response : client.getResponseModels()) {
+                javaPackage.addClientResponse(response.getPackage(), response.getName(), response);
+            }
+
+            // Client model
+            for (ClientModel model : client.getModels()) {
+                javaPackage.addModel(model.getPackage(), model.getName(), model);
+            }
+
+            // Enum
+            for (EnumType enumType : client.getEnums()) {
+                javaPackage.addEnum(enumType.getPackage(), enumType.getName(), enumType);
+            }
+
+            // XML sequence wrapper
+            for (XmlSequenceWrapper xmlSequenceWrapper : client.getXmlSequenceWrappers()) {
+                javaPackage.addXmlSequenceWrapper(xmlSequenceWrapper.getPackage(),
+                        xmlSequenceWrapper.getWrapperClassName(), xmlSequenceWrapper);
+            }
+
+            // Exception
+            for (ClientException exception : client.getExceptions()) {
+                javaPackage.addException(exception.getPackage(), exception.getName(), exception);
+            }
+
+            // Package-info
+            for (PackageInfo packageInfo : client.getPackageInfos()) {
+                javaPackage.addPackageInfo(packageInfo.getPackage(), "package-info", packageInfo);
+            }
+
+            // TODO: POM, Manager

```

Comment:
Pom could be handled by the vanilla gen - we can discuss. BTW I wrote an xml DSL util just like the Java one, which could be useful

Reply:
Sure. Fluent does not need pom at this stage. Fluent lite will need the POM but that will be months later.